Makefile.psp.mk

```diff
@@ -1,123 +1,53 @@
-
-extract_pspeu: $(addprefix $(BUILD_DIR)/,$(addsuffix .ld,$(PSP_EU_EXTRACT_TARGETS)))
-
-$(WIBO):
-	wget -O $@ https://github.com/decompals/wibo/releases/download/0.6.13/wibo
-	sha256sum --check $(WIBO).sha256
-	chmod +x $(WIBO)
-$(MWCCPSP): $(WIBO) $(BIN_DIR)/mwccpsp_219
+# Step 1/2 of extract
+$(BUILD_DIR)/%.ld: $(CONFIG_DIR)/splat.$(VERSION).%.yaml $(BASE_SYMBOLS) $(CONFIG_DIR)/symbols.$(VERSION).%.txt | $(EXTRACTED_DISK_DIR) $(VENV_DIR)
+	$(muffle)$(SPLAT) $<
 
-$(MWCCGAP_APP):
-	git submodule init $(MWCCGAP_DIR)
-	git submodule update $(MWCCGAP_DIR)
+# Step 2/2 of extract
+extract_pspeu: $(addprefix $(BUILD_DIR)/,$(addsuffix .ld,$(call get_targets,prefixed)))
 
-dra_psp: $(BUILD_DIR)/dra.bin
-ric_psp: $(BUILD_DIR)/ric.bin
-stlib_psp: $(BUILD_DIR)/lib.bin
-stno4_psp: $(BUILD_DIR)/no4.bin
-stst0_psp: $(BUILD_DIR)/st0.bin
-stwrp_psp: $(BUILD_DIR)/wrp.bin
-tt_000_psp: $(BUILD_DIR)/tt_000.bin
-
-$(BUILD_DIR)/%.bin: $(BUILD_DIR)/%.elf
-	$(OBJCOPY) -O binary $< $@
-$(BUILD_DIR)/lib.bin: $(BUILD_DIR)/stlib.elf
-	$(OBJCOPY) -O binary $< $@
-$(BUILD_DIR)/no4.bin: $(BUILD_DIR)/stno4.elf
-	$(OBJCOPY) -O binary $< $@
-$(BUILD_DIR)/st0.bin: $(BUILD_DIR)/stst0.elf
-	$(OBJCOPY) -O binary $< $@
-$(BUILD_DIR)/wrp.bin: $(BUILD_DIR)/stwrp.elf
-	$(OBJCOPY) -O binary $< $@
-
-$(BUILD_DIR)/dra.ld: $(CONFIG_DIR)/splat.pspeu.dra.yaml $(BASE_SYMBOLS) $(CONFIG_DIR)/symbols.pspeu.dra.txt
-	$(SPLAT) $<
-$(BUILD_DIR)/ric.ld: $(CONFIG_DIR)/splat.pspeu.ric.yaml $(BASE_SYMBOLS) $(CONFIG_DIR)/symbols.pspeu.ric.txt
-	$(SPLAT) $<
-$(BUILD_DIR)/st%.ld: $(CONFIG_DIR)/splat.pspeu.st%.yaml $(BASE_SYMBOLS) $(CONFIG_DIR)/symbols.pspeu.st%.txt
-	$(SPLAT) $<
-$(BUILD_DIR)/tt_%.ld: $(CONFIG_DIR)/splat.pspeu.tt_%.yaml $(BASE_SYMBOLS) $(CONFIG_DIR)/symbols.pspeu.tt_%.txt
-	$(SPLAT) $<
-
-$(BUILD_DIR)/dra.elf: $(BUILD_DIR)/%.elf: $(BUILD_DIR)/dra.ld $$(call get_merged_o_files,%) $$(call list_o_files_psp,dra_psp)
-	$(call link,dra,$@)
-
-$(BUILD_DIR)/ric.elf: $(BUILD_DIR)/%.elf: $(BUILD_DIR)/ric.ld $$(call get_merged_o_files,%) $$(call list_o_files_psp,ric_psp) $(BUILD_DIR)/assets/ric/mwo_header.bin.o
-	$(call link,ric,$@)
-
-$(BUILD_DIR)/tt_%.elf: $(BUILD_DIR)/tt_%.ld $$(call list_o_files_psp,servant/tt_$$*) $(BUILD_DIR)/assets/servant/tt_%/mwo_header.bin.o
+# Step 1/5 of build
+$(BUILD_DIR)/%.s.o: %.s $(AS)
+	$(call echo,Assembling $<,optional) mkdir -p $(dir $@)
+	$(muffle)$(AS) $(AS_FLAGS) -o $@ $<
+$(BUILD_DIR)/%.c.o: %.c $(MWCCPSP) $(MWCCGAP_APP) $(AS) | $(VENV_DIR)/bin
+	$(call echo,Compiling $<,optional) mkdir -p $(dir $@)
+	$(muffle)$(SOTNSTR_APP) process -p -f $< | $(PYTHON) $(MWCCGAP_APP) $@ --src-dir $(dir $<) $(COMPILER_ARGS)
+$(BUILD_DIR)/assets/%/mwo_header.bin.o: assets/%/mwo_header.bin
+	$(call echo,Building $@,optional) mkdir -p $(dir $@)
+	$(muffle)$(LD) -r -b binary -o $@ $<
+
+# Step 2/5 of build
+$(foreach target,$(filter-out main,$(GAME)),$(BUILD_DIR)/$(target).elf): $(BUILD_DIR)/%.elf: $(BUILD_DIR)/%.ld $$(call get_psp_o_files,%)
```

Comment:
Why are we explicitly removing `main`? There is the main executable `ps.elf` in the PSP build that might be added at some point

Reply:
This was a carryover from psx where main is handled a little differently than the others in that group.   I had thought I had removed the filter from psp, but I guess not.